lib/go-tc/deliveryservice_requests.go

```diff
@@ -20,15 +20,394 @@ import (
+		validation.Field(&d.EmailTo, validation.Required, is.Email),
+	)
+	if err != nil {
+		errs = append(errs, err)
+	}
+
+	details := d.Details
+	err = validation.ValidateStruct(&details,
+		validation.Field(&details.ContentType, validation.Required),
+		validation.Field(&details.Customer, validation.Required),
+		validation.Field(&details.DeepCachingType, validation.By(
+			func(t interface{}) error {
+				if t == nil {
+					return errors.New("deepCachingType: required")
+				}
+				if *t.(*DeepCachingType) == DeepCachingTypeInvalid {
+					return errors.New("deepCachingType: invalid Deep Caching Type")
+				}
+				return nil
+			})),
+		validation.Field(&details.DeliveryProtocol, validation.By(
+			func(p interface{}) error {
+				if p == nil {
+					return errors.New("deliveryProtocol: required")
+				}
+				if *p.(*Protocol) == ProtocolInvalid {
+					return errors.New("deliveryProtocol: invalid Protocol")
+				}
+				return nil
+			})),
+		validation.Field(&details.HasNegativeCachingCustomization, validation.By(
+			func (h interface{}) error {
+				if h == nil {
+					return errors.New("hasNegativeCachingCustomization: required")
+				}
+				return nil
+			})),
+		validation.Field(&details.HasOriginACLWhitelist, validation.By(
+			func (h interface{}) error {
+				if h == nil {
+					return errors.New("hasNegativeCachingCustomization: required")
+				}
+				return nil
+			})),
+		validation.Field(&details.HasOriginDynamicRemap, validation.By(
+			func (h interface{}) error {
+				if h == nil {
+					return errors.New("hasNegativeCachingCustomization: required")
+				}
+				return nil
+			})),
+		validation.Field(&details.HasSignedURLs, validation.By(
+			func (h interface{}) error {
+				if h == nil {
+					return errors.New("hasNegativeCachingCustomization: required")
+				}
+				return nil
+			})),
+		validation.Field(&details.MaxLibrarySizeEstimate, validation.Required),
+		validation.Field(&details.OriginHeaders, validation.By(
+			func (h interface{}) error {
+				if h == nil {
+					return nil
+				}
+				if len(*h.(*OriginHeaders)) < 1 {
+					return errors.New("originHeaders: cannot be an empty list (use 'null' if none)")
+				}
+				return nil
+			})),
+		validation.Field(&details.OriginTestFile, validation.Required),
+		validation.Field(&details.OriginURL, validation.Required, is.URL),
+		validation.Field(&details.PeakBPSEstimate, validation.Required),
+		validation.Field(&details.PeakTPSEstimate, validation.Required),
+		validation.Field(&details.QueryStringHandling, validation.Required),
+		validation.Field(&details.RangeRequestHandling, validation.Required),
+		validation.Field(&details.RoutingName, validation.Required),
+		validation.Field(&details.RoutingType, validation.By(
+			func (t interface{}) error {
+				if t == nil || *(t.(*DSType)) == "" {
+					return errors.New("routingType: required")
+				}
+				*t.(*DSType) = DSTypeFromString(string(*t.(*DSType)))
+				switch *(t.(*DSType)) {
```

Comment:
Instead of this switch statement can we simply check if the result from `DSTypeFromString` is `DSTypeInvalid` or not? That function basically already contains this switch statement.